modules/core/src/minmax.cpp

```diff
@@ -71,33 +71,760 @@ minMaxIdx_( const T* src, const uchar* mask, WT* _minVal, WT* _maxVal,
+                        idxMin = v_select(cmpMin, idx, idxMin);
+                        idxMax = v_select(cmpMax, idx, idxMax);
+                        valMin = v_select(v_reinterpret_as_f32(cmpMin), data, valMin);
+                        valMax = v_select(v_reinterpret_as_f32(cmpMax), data, valMax);
+                        idx += inc;
+                        data = v_load(src + k + v_float32x4::nlanes);
+                        cmpMin = v_reinterpret_as_u32(v_reinterpret_as_s32(data < valMin) & maskVal2);
+                        cmpMax = v_reinterpret_as_u32(v_reinterpret_as_s32(data > valMax) & maskVal2);
+                        idxMin = v_select(cmpMin, idx, idxMin);
+                        idxMax = v_select(cmpMax, idx, idxMax);
+                        valMin = v_select(v_reinterpret_as_f32(cmpMin), data, valMin);
+                        valMax = v_select(v_reinterpret_as_f32(cmpMax), data, valMax);
+                        idx += inc;
+                    }
+                }
+
+                j = k;
+
+                minMaxIdx_reduce_f32( valMin, valMax, idxMin, idxMax, none, minVal, maxVal,
+                                      minIdx, maxIdx, delta );
+            }
+            while ( j < len0 );
+        }
+
+        minMaxIdx_finish( src, mask, minval, maxval, minidx, maxidx, minVal, maxVal,
+                          minIdx, maxIdx, len, startidx, j );
+    }
+    else
+    {
+        minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx);
+    }
+#else
+    minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx );
+#endif
+}
 
 static void minMaxIdx_64f(const double* src, const uchar* mask, double* minval, double* maxval,
                           size_t* minidx, size_t* maxidx, int len, size_t startidx )
-{ minMaxIdx_(src, mask, minval, maxval, minidx, maxidx, len, startidx ); }
+{
+#if CV_SIMD128_64F
+    if ( len >= 4 * v_float64x2::nlanes )
+    {
+        int j, len0;
+        double minVal, maxVal;
+        size_t minIdx, maxIdx;
+
+        minMaxIdx_init( src, mask, minval, maxval, minidx, maxidx, minVal, maxVal, minIdx, maxIdx,
+                        DBL_MIN, DBL_MAX, 4 * v_float64x2::nlanes, len, startidx, j, len0 );
+
+        if ( len0 - j >= 4 * v_float64x2::nlanes )
+        {
+            v_uint64x2 inc = v_setall_u64(v_float64x2::nlanes);
+            v_uint64x2 none = v_reinterpret_as_u64(v_setall_s64(-1));
+            v_uint64x2 idxStart(0, 1);
+
+            do
+            {
+                v_float64x2 valMin = v_setall_f64(minVal), valMax = v_setall_f64(maxVal);
+                v_uint64x2 idx = idxStart, idxMin = none, idxMax = none;
+
+                int k = j;
+                size_t delta = startidx + j;
+
+                if ( !mask )
+                {
+                    for( ; k < std::min(len0, j + 32764 * 4 * v_float64x2::nlanes); k += 4 * v_float64x2::nlanes )
```

Comment:
Why do we need x4 unrolling?

Reply:
It makes it a lot easier to handle the 'mask' case down below (v_load_expand_q is really bad on some platforms).

Reply:
This loop doesn't work this `mask`.

Perhaps, loop below can be iterated over loaded `mask` size:
```.diff
-4 * v_float64x2::nlanes
+v_uint16x8::nlanes
```
with corresponding comment near "for" that loop iterations are performed over `mask` values.

Reply:
I'd rather keep it the way it is for consistency with the other cases.